### PR TITLE
Issue 10: Additional context around the text span for NLP

### PR DESCRIPTION
**Pull request by mcbride-p**
Code changes to show additional context (up to 30 characters) around the text span for NLP and highlight the actual span

### CODE REVIEW

File: src/components/Fhir/InsightsDetailButton.js [hunk 1]
```diff
@@ -76,7 +77,36 @@ export default class InsightsDetailButton extends React.Component {
...
+                                        ? "..." +
+                                        details.evidenceDetail.substring(
+                                        details.offsetBegin - extendedCharacters,
+                                        details.offsetBegin
+                                        )
+                                        : details.evidenceDetail.substring(
+                                        0,
+                                        details.offsetBegin
+                                        )}
+                                        <span class="search-match">
```
Comment (cragun47) [inline src/components/Fhir/InsightsDetailButton.js:92]: Shouldn't this be `className`?

File: src/components/Fhir/InsightsDetailButton.js [hunk 2]
```diff
@@ -76,7 +77,36 @@ export default class InsightsDetailButton extends React.Component {
                                 <tbody>
                                     <tr>
                                         <td><strong>Covered Text</strong></td>
-                                        <td>"...<em>{details.coveredText}</em>..." [{details.offsetBegin}:{details.offsetEnd}]</td>
+                                        <td>
+                                        "
```
Comment (cragun47) [inline src/components/Fhir/InsightsDetailButton.js:81]: Formatting would greatly help the understandability of the code segments before and after the span.

**Review by cragun47 — APPROVED** [review]
Couple of comments to consider.

**Review by ewill00 — APPROVED** [review]
A few questions:
This is to extend 30 characters on both sides, is that correct?  No need to change it now but is there a way to configure that in the future or should we just change the const if we want to try different amounts?
Does it handle the case where the +30 or -30 takes you past the beginning or end of the text without error?  (ie a substring of a negative start ok or is that an error)
It does nothing to try to be on a word boundary, correct?  I think that is ok with 30 on each side but wanted to understand if I was missing it.
Thanks!

File: src/components/Fhir/InsightsDetailButton.js (hunk 1 above)
Comment (mcbride-p) [reply src/components/Fhir/InsightsDetailButton.js:92]: Either works but className is best practice so I'll change it.

File: src/components/Fhir/InsightsDetailButton.js (hunk 2 above)
Comment (mcbride-p) [reply src/components/Fhir/InsightsDetailButton.js:81]: Sorry... not sure how it got checked in like that. I'll sort it before merging.

**Comment by mcbride-p** [comment]
@ewill00 - To answer your questions above ...
Yes const value determines character extension on either side and agree this should be configurable and determined from a config file that can be changed on deployment going forward. I'll create ticket for that.
It does handle scenarios where there are in excess of 30 characters before and after the startoffset and endoffset respectively. In either case span text is preceded/superceded by three dots after the 30 characters to indicate evidence_detail text extends further.
Correct - No we dont handle ending extended text on word boundary.

**Pull request merged**